packages/cli/prisma/schema.prisma

```diff
@@ -39,3 +39,56 @@ model ApiKey {
 
   @@index([organizationId])
 }
+
+model Message {
+  // A unique email message delivered to one or multiple users.
+  id        String   @id @default(cuid())
+  createdAt DateTime @default(now())
+  to        String[]
+  cc        String[]
+  bcc       String[]
+  subject   String?
+  from      String?
+
+  // Open related fields
+  openedAt  DateTime?
+  openCount Int       @default(0)
+
+  // Mailing specific metadata
+  templateName String? // reference to a mailing template
+  previewName  String? // reference to a mailing preview
+
+  Click          Click[]
+  MessageContent MessageContent?
+
+  @@index([templateName, previewName])
+}
+
+model MessageContent {
+  // Note: This table will likely grow large over time, we will likely
+  // want to truncate it in the future to only retain last N messages.
+  // We also might need to consider moving it to an external data store like S3.
+  id        String   @id @default(cuid())
+  createdAt DateTime @default(now())
+  html      String?
+  messageId String
+  Message   Message  @relation(fields: [messageId], references: [id])
```

Comment:
@psugihara @alexfarrill Just a heads up, I switched where the relation is stored for this. In the originally agreed upon schema the relation was stored on message (well technically Send). I flipped it so that if/when we truncate the messagecontent table, we'll be able to keep referential integrity. Does that make sense?

Reply:
👍 yep seems good